### ISSUE TITLE
Backtesting trading strategy performance

### ISSUE THREAD

**Issue by ghost** [tg12/FAIG#10]
Any idea on backtesting?

**Comment by tg12** [comment]
For back-testing, We would need to get some historical data.... This would be VERY heavy on the API usage. Perhaps we could get this into some kind of database?

**Comment by ghost** [comment]
Yea, the market data fetched is supposed to be saved for later use.

How about keeping this issue opened, and wait until later when the project structure is better understood then we start making this saving function?

**Comment by tg12** [comment]
Yeah can do!

**Comment by FlickeringIRoam** [comment]
I've been thinking about this for a little while.  I don't want to use a central-DB due to open read/write permissions, and while that'd help data sharing, I suspect would break IG's T&Cs.  So something local.  sqlite3 is near-perfect, but sql is a pain once you've experienced nosql done well, and this project _screams_ nosql.  However, it's also burdensome to expect people to setup mongodb or similar, and we really don't need the performance yet, so i think tinydb http://tinydb.readthedocs.io/en/latest/intro.html or similar is the way to go.  I was going to implement this for market data until i realised that market dealhandling rules CAN change per-request, and don't count against the history api limit anyway.

**Comment by callamd** [comment]
@FlickeringIRoam why not consider dockerising the application, and running a database designed for timeseries like InfluxDB? Thus the entire stack could be bought up with `docker-compose`.

**Comment by FlickeringIRoam** [comment]
Yep, very valid point. I have no issues at all with distributing it as a
container

**Comment by tg12** [comment]
Closing due to inactivity — if this is still relevant please feel free to re-open.

**Comment by DICAP1** [comment]
Hey everyone,

Just wanted to say I’ve been following the work on FAIG for a bit now—it’s a seriously clean, well-structured framework for automating execution against the IG API. Finding solid open-source tools for this space isn't easy, so hats off to you guys.

I'm reaching out because a few of us have been building an intelligence-driven pre-trade filter called Sovereign (deepalgo.co.uk). It essentially tracks real-time market regimes (via Hidden Markov Models) and order flow toxicity (VPIN) right at the liquidity layer to catch sudden flash crashes or toxic spread-widening before they hit your account.

We realized it slots perfectly into setups like FAIG as a quick, non-blocking middleware check right before your code fires a trade out to the broker. Something like:

Python
# Quick safety check before pinging IG
headers = {"X-DeepAlgo-Key": "YOUR_BETA_KEY"}
res = requests.post("https://api.deepalgo.co.uk/v4/approve", json={"instrument": "EUR_USD"}, headers=headers)

if res.json().get("status") == "APPROVED":
    ig_client.place_order(...) # Trade safely
else:
    logger.warning("Sovereign triggered a VETO — standing down to protect capital.")


We’re running an early beta right now and would love to get some real, automated quants from this community to break it, test the endpoints, and see how it holds up. We're happy to hand out free beta keys seeded with 10,000 API hits so you can mess around with it in your own dev environments without worrying about limits.

If anyone wants to grab a key and a copy of the endpoint docs to play with, just shoot us a quick email at support@deepalgo.co.uk and we'll get you set up.

Keen to hear what you folks think!
